blog/content/post/2024-10-11-uefi-capsule-updates.md

```diff
@@ -0,0 +1,399 @@
+           support UEFI capsule updates. The additional way to update your
+           firmware aims at making the process more reliable and convenient.'
+cover: /covers/capsule-updates.png
+author: sergii.dmytruk
+layout: post
+published: true
+date: 2024-10-11
+archives: "2024"
+
+tags:
+  - coreboot
+  - edk2
+  - firmware
+  - uefi
+categories:
+  - Firmware
+
+---
+
+Dasharo firmware starts to support capsule updates.  This post looks into what
+it is, why it is useful, and what has led to this.
+
+## Capsule updates for MSI boards
+
+The first boards to get the support are [MSI PRO Z690-A (v1.1.4)][z690] and [MSI
+PRO Z790-P (v0.9.2)][z790] (all their WIFI/DDR4/DDR5 variants).  Do note that
+the functionality cannot be used to update from previous firmware releases
+for these boards.  Only future updates can leverage this update method unless
+you use it to "reinstall" newest releases as an experiment.
+
+Worth mentioning right away that this work was funded by [NLnet
+Foundation][nlnet] in the interest of making updates of open source firmware a
+more pleasant experience.  More details about the project can be found
+[here][nlnet-project].
+
+[z690]: https://docs.dasharo.com/variants/msi_z690/releases/#v114-2024-
+[z790]: https://docs.dasharo.com/variants/msi_z790/releases/#v092-2024-
+[nlnet]: https://nlnet.nl/
+[nlnet-project]: https://docs.dasharo.com/projects/capsule-updates/
+
+## Firmware updates
+
+Firmware is a crucial software component without which a device can't function
+properly, in fact, most likely can't function at all.  Like anything, firmware
+is subject to defects and, due to its integral role for a device, a firmware
+defect can have a significant impact on its operation.  A bad firmware can be
+subject to an attack or sometimes cause a physical damage to a device.  Even in
+the absence of any issues, an update might still be desirable to provide more
+configuration options or support wider range of hardware.
+
+For these and other reasons firmware updates are a necessity.  As devices
+get more advanced, firmware defects become more likely, thus further increasing
+the importance of keeping firmware up-to-date.
+
+Unfortunately, carrying out a firmware update is a relatively difficult task
+that itself can go wrong and lead to a broken device.  This is hard to address
+in full, but some update methods are more user-friendly than others and update
+capsules are an example of moving in the direction of greater convenience to
+end users.
+
+## Firmware update methods
+
+Let's have a quick overview of update methods to have the necessary context for
+discussing capsule updates.
+
+First of all, an update can be performed either internally or externally.  An
+external update often requires a special device (see [this kit][ec-kit] for an
+idea) capable of updating a [flash chip] which contains the firmware.  The
+device gets connected to a host machine (so you need a second PC/lapatop) from
```

Comment:
```suggestion
device gets connected to a host machine (so you need a second PC/laptop) from
```

Reply:
https://github.com/3mdeb/news-and-ideas/compare/fc56b56081baa6eafafea7e60a43adde55d7283b..93c8836df2ed0f57e65ae72375ebc06e04d2b5b2